src/full/Agda/Interaction/Library/Parse.hs

```diff
@@ -56,7 +56,7 @@ runP :: P a -> (Either String a, [LibWarning'])
 runP = runWriter . runExceptT
 
 -- | Library Warnings.
-data LibWarning'
+newtype LibWarning'
   = UnknownField String
   deriving (Show, Data)
```

Comment:
Semantically, this is not a newtype.

Reply:
I just looked for cases that appeared to effectively be type synonyms. Are there any that pass your semantic test?

Reply:
> Semantically, this is not a newtype.

What do you mean? Do you need the extra bottom?

Reply:
I mean that it's not just a synonym for `String`, it's a proper enumeration
that happens to have one case at the moment but will probably get extended
in the future as we add more warnings / suppress more errors.

Reply:
Is there a consensus on whether this PR is worth pursuing?  Sounds as if it might not be, as resolution depends on judging whether a data declaration is a candidate for extension in future.

Reply:
The performance impact of this choice is likely to be minimal.

Reply:
I haven't tested it for performance.  What is the easiest way of doing that?

Reply:
My point is that we're wasting time discussing something that most likely does not matter.